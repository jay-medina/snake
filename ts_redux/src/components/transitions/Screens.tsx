import React from 'react';
import { PlayButton } from './PlayButton';

export interface StartProps {
  onPlayClick(): void;
}

export function Start({ onPlayClick }: StartProps) {
  return (
    <div className="snake__start-screen">
      <div className="snake__start-screen-title">Snake</div>
      <PlayButton onClick={onPlayClick} />
    </div>
  );
}

export function GameOver({ onPlayClick }: StartProps) {
  return (
    <div className="snake__game-over-screen">
      <div className="snake__game-over-title">Game Over</div>
      <PlayButton onClick={onPlayClick} />
    </div>
  );
}
