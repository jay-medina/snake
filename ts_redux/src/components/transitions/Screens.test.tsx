import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { GameOver, Start, StartProps } from './Screens';

describe('<Start />', () => {
  let onPlayClick: jest.Mock;
  let wrapper: ShallowWrapper<StartProps>;

  beforeEach(() => {
    onPlayClick = jest.fn();
    wrapper = shallow(<Start onPlayClick={onPlayClick} />);
  });

  afterEach(() => {
    wrapper && wrapper.unmount();
  });

  it('renders the component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('when play button is clicked', () => {
    it('invokes the onPlayClick callback', () => {
      wrapper.find('PlayButton').simulate('click');
      expect(onPlayClick).toHaveBeenCalled();
    });
  });
});

describe('<GameOver />', () => {
  let onPlayClick: jest.Mock;
  let wrapper: ShallowWrapper<StartProps>;

  beforeEach(() => {
    onPlayClick = jest.fn();
    wrapper = shallow(<GameOver onPlayClick={onPlayClick} />);
  });

  afterEach(() => {
    wrapper && wrapper.unmount();
  });

  it('renders the component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('when play button is clicked', () => {
    it('invokes the onPlayClick callback', () => {
      wrapper.find('PlayButton').simulate('click');
      expect(onPlayClick).toHaveBeenCalled();
    });
  });
});
