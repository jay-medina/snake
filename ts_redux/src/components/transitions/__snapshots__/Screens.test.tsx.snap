// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameOver /> renders the component 1`] = `
<div
  className="snake__game-over-screen"
>
  <div
    className="snake__game-over-title"
  >
    Game Over
  </div>
  <PlayButton
    onClick={[MockFunction]}
  />
</div>
`;

exports[`<Start /> renders the component 1`] = `
<div
  className="snake__start-screen"
>
  <div
    className="snake__start-screen-title"
  >
    Snake
  </div>
  <PlayButton
    onClick={[MockFunction]}
  />
</div>
`;
